tests/includes/transformer/class-test-factory.php

```diff
@@ -214,4 +214,53 @@ function () {
 
 		remove_all_filters( 'activitypub_transformer' );
 	}
+
+	/**
+	 * Test successful URI transformation.
+	 */
+	public function test_successful_uri_transformation() {
+		// Mock-Daten für die HTTP-Antwor;
+		$fake_request = function () {
+			return array(
+				'response' => array( 'code' => 200 ),
+				'body'     => json_encode(
+					array(
+						'id' => 'https://example.com/activity/1',
+						'type' => 'Note',
+						'content' => 'Test Content',
+					),
+				),
+			);
+		};
+
+		add_filter( 'pre_http_request', $fake_request, 10 );
+
+		$uri_transformer = Factory::get_transformer( 'https://example.com/activity/1' );
+		$result = $uri_transformer->to_object();
+
+		$this->assertIsObject( $result );
+		$this->assertEquals( 'https://example.com/activity/1', $result->get_id() );
+		$this->assertEquals( 'Note', $result->get_type() );
+		$this->assertEquals( 'Test Content', $result->get_content() );
+
+		remove_filter( 'pre_http_request', $fake_request, 10 );
+	}
+
+	/**
+	 * Test URI transformation with error.
+	 */
+	public function test_uri_transformation_error() {
+		// WP_Error für fehlgeschlagene Anfrage erstellen
+		$fake_request = function () {
+			return new \WP_Error( 'fetch_error', 'Failed to fetch remote object' );
+		};
+
+		add_filter( 'pre_http_request', $fake_request, 10 );
+
+		$uri_transformer = Factory::get_transformer( 'https://example.com/invalid' );
+
+		$this->assertInstanceOf( \WP_Error::class, $uri_transformer );
```

Comment:
No need to change this, but you recently mentioned how much you enjoy learning new unit test functions: WordPress created [`assertWPError()`](https://github.com/WordPress/wordpress-develop/blob/381731aafc815c399b13a524c519fc1c3fe00018/tests/phpunit/includes/abstract-testcase.php#L837) as a wrapper for the `assertInstanceOf()` check :)